tests/test_notebooks.py

```
@@ -0,0 +1,21 @@
+import subprocess
+import glob
+import pytest
+
+
+@pytest.mark.parametrize("notebook", glob.glob("**/*.ipynb", recursive=True))
+def test_notebook_executes(notebook):
+    """Test that notebook executes without errors."""
+    if notebook.startswith('.'):
+        pytest.skip("Skipping hidden notebook")
+
+    result = subprocess.run([
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)